### PR TITLE
_linregress_ufunc handling of nans

### PR DESCRIPTION
**Pull request by hogikyan**
added 'mask' feature to _linregress_ufunc in utils.py so that you can give nans to xr_linregress
slope etc. will all be nan for regressions without non-nan points

### CODE REVIEW

**Comment by jbusecke** [comment]
I like the idea of this option, but I am not sure about the implementation. 
This adds a lot of code that is not optional (e.g. all the computations are always executed, even if there are no nans in the input object).
I think we should implement this with a keyword.

File: xarrayutils/utils.py [hunk 1]
```diff
@@ -61,7 +68,7 @@ def xr_linregress(a, b, dim='time', convert_to_dataset=True, dtype=None):
 
     Parameters
     ----------
-    a : {xr.DataArray}
+    a : {xr.DataArray} (does it have to be tho? my 'a' is vector)
```
Comment (jbusecke) [inline xarrayutils/utils.py]: That is just the type of the input. The data can be 1 dimensional (which is what I assume you mean with 'vector') or n-dimensional. http://xarray.pydata.org/en/stable/generated/xarray.DataArray.html

Comment (hogikyan) [reply xarrayutils/utils.py]: oops forgot I left that note sorry!

**Comment by jbusecke** [comment]
This has been implemented as an option now. You can do `xr_linregress(..., nanmask=True)` (it is off by default to not change older code. Still need to write some tests for it. Will close this here. Sorry for the long turnaround

**Pull request closed without merging**